### PR TITLE
tests: clean-up functests by using libvmi factory functions

### PR DESCRIPTION
**Pull request by acardace**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: tests/container_disk_test.go [hunk 1]
```diff
@@ -193,7 +193,7 @@ var _ = Describe("[rfe_id:588][crit:medium][vendor:cnv-qe@redhat.com][level:comp
 	Describe("[rfe_id:273][crit:medium][vendor:cnv-qe@redhat.com][level:component]Starting from custom image location", func() {
 		Context("with disk at /custom-disk/downloaded", func() {
 			It("[test_id:1466]should boot normally", func() {
-				vmi := tests.NewRandomVMIWithEphemeralDiskAndUserdata(cd.ContainerDiskFor(cd.ContainerDiskCirrosCustomLocation), "#!/bin/bash\necho 'hello'\n")
+				vmi := libvmi.NewCirros()
```
Comment (EdDev) [inline tests/container_disk_test.go:196]: This is using a different image: `cd.ContainerDiskCirrosCustomLocation`.
You should try `libvmi.NewCirros(WithContainerImage(cd.ContainerDiskFor(cd.ContainerDiskCirrosCustomLocation)))`.

**Review by EdDev — CHANGES_REQUESTED** [review]
Nice, thanks!
There is one change I am unsure if you intended for (see inline).

File: tests/libvmi/vmi.go [hunk 2]
```diff
@@ -36,7 +37,7 @@ type Option func(vmi *kvirtv1.VirtualMachineInstance)
 // New instantiates a new VMI configuration,
 // building its properties based on the specified With* options.
 func New(name string, opts ...Option) *kvirtv1.VirtualMachineInstance {
-	vmi := baseVmi(name)
+	vmi := baseVmi(util.NamespaceTestDefault, name)
```
Comment (xpivarc) [inline tests/libvmi/vmi.go:40]: This should not be needed.

File: tests/libvmi/vmi.go [hunk 3]
```diff
@@ -125,8 +126,15 @@ func WithSEV() Option {
...
-	vmi := kvirtv1.NewVMIReferenceFromNameWithNS("", name)
+// WithNamespace configures the namespace in which the VMI will be created.
+func WithNamespace(namespace string) Option {
+	return func(vmi *kvirtv1.VirtualMachineInstance) {
+		vmi.Namespace = namespace
+	}
+}
+
+func baseVmi(namespace, name string) *kvirtv1.VirtualMachineInstance {
+	vmi := kvirtv1.NewVMIReferenceFromNameWithNS(namespace, name)
```
Comment (xpivarc) [inline tests/libvmi/vmi.go:137]: The namespace should not be set. Can you clarify why we need it?

File: tests/vmi_lifecycle_test.go [hunk 4]
```diff
@@ -1549,7 +1545,7 @@ var _ = Describe("[rfe_id:273][crit:high][arm64][vendor:cnv-qe@redhat.com][level
 		})
 		Context("with ACPI and some grace period seconds", func() {
 			DescribeTable("[rfe_id:273][crit:medium][vendor:cnv-qe@redhat.com][level:component]should result in vmi status succeeded", func(gracePeriod int64) {
-				vmi = newCirrosVMI()
+				vmi = libvmi.NewCirros()
```
Comment (xpivarc) [inline tests/vmi_lifecycle_test.go:1548]: Maybe you could also include the gracePeriod option.

File: tests/libvmi/vmi.go [hunk 5]
```diff
@@ -125,8 +126,8 @@ func WithSEV() Option {
 	}
 }
 
-func baseVmi(name string) *kvirtv1.VirtualMachineInstance {
-	vmi := kvirtv1.NewVMIReferenceFromNameWithNS("", name)
+func baseVmi(namespace, name string) *kvirtv1.VirtualMachineInstance {
```
Comment (EdDev) [inline tests/libvmi/vmi.go:136]: The VMI spec should not set the namespace.
This is a repeating mistake.

Tge namespace is to br set by the creation call, which will return a vmi that includes the ns.

**Review by EdDev — CHANGES_REQUESTED** [review]

File: tests/container_disk_test.go [hunk 6]
```diff
@@ -193,7 +193,11 @@ var _ = Describe("[rfe_id:588][crit:medium][vendor:cnv-qe@redhat.com][level:comp
 	Describe("[rfe_id:273][crit:medium][vendor:cnv-qe@redhat.com][level:component]Starting from custom image location", func() {
 		Context("with disk at /custom-disk/downloaded", func() {
 			It("[test_id:1466]should boot normally", func() {
-				vmi := tests.NewRandomVMIWithEphemeralDiskAndUserdata(cd.ContainerDiskFor(cd.ContainerDiskCirrosCustomLocation), "#!/bin/bash\necho 'hello'\n")
+				vmi := libvmi.NewCirros(
```
Comment (xpivarc) [inline tests/container_disk_test.go:196]: I am not sure this will work as you expect. Have a look at the `WithContainerImage` option.

**Review by xpivarc — CHANGES_REQUESTED** [review]

File: tests/libvmi/vmi.go (hunk 5 above)
Comment (acardace) [reply tests/libvmi/vmi.go:136]: gotcha, I did this because some tests started failing because the VMI had no namespace set, I guess they relied on this.

I'll fix this.

https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/7551/pull-kubevirt-e2e-k8s-1.21-sig-compute/1513783827359600640

File: tests/libvmi/vmi.go (hunk 2 above)
Comment (acardace) [reply tests/libvmi/vmi.go:40]: https://github.com/kubevirt/kubevirt/pull/7551#discussion_r849195120

File: tests/libvmi/vmi.go (hunk 3 above)
Comment (acardace) [reply tests/libvmi/vmi.go:137]: https://github.com/kubevirt/kubevirt/pull/7551#discussion_r849195120

File: tests/vmi_lifecycle_test.go (hunk 4 above)
Comment (acardace) [reply tests/vmi_lifecycle_test.go]: done

File: tests/container_disk_test.go (hunk 6 above)
Comment (acardace) [reply tests/container_disk_test.go]: fixed it

**Comment by acardace** [comment]
I have to fix some more e2e tests that started failing after switching to libvmi because of the assumption that the namespace is already populated in the object prior to creation.

File: tests/security_features_test.go [hunk 7]
```diff
@@ -67,7 +67,7 @@ var _ = Describe("[Serial][sig-compute]SecurityFeatures", func() {
 				config.SELinuxLauncherType = "container_t"
 				tests.UpdateKubeVirtConfigValueAndWait(*config)
 
-				vmi = tests.NewRandomVMIWithEphemeralDiskAndUserdata(cd.ContainerDiskFor(cd.ContainerDiskCirros), "#!/bin/bash\necho 'hello'\n")
+				vmi := libvmi.NewCirros()
```
Comment (xpivarc) [inline tests/security_features_test.go:70]: Is the `:=` correct?

File: tests/vmi_lifecycle_test.go [hunk 8]
```diff
@@ -1549,24 +1560,24 @@ var _ = Describe("[rfe_id:273][crit:high][arm64][vendor:cnv-qe@redhat.com][level
...
-				vmi = newCirrosVMI()
+				var vmi *v1.VirtualMachineInstance
 
 				if gracePeriod >= 0 {
-					vmi.Spec.TerminationGracePeriodSeconds = &gracePeriod
+					vmi = libvmi.NewCirros(libvmi.WithTerminationGracePeriod(gracePeriod))
 				} else {
 					gracePeriod = v1.DefaultGracePeriodSeconds
-					vmi.Spec.TerminationGracePeriodSeconds = nil
+					vmi = libvmi.NewCirros()
```
Comment (xpivarc) [inline tests/vmi_lifecycle_test.go:1569]: Is this the same?

File: tests/migration_test.go [hunk 9]
```diff
@@ -3095,7 +3101,7 @@ var _ = Describe("[rfe_id:393][crit:high][vendor:cnv-qe@redhat.com][level:system
 
 					vmi_evict1 := alpineVMIWithEvictionStrategy()
 					vmi_evict2 := alpineVMIWithEvictionStrategy()
-					vmi_noevict := tests.NewRandomVMIWithEphemeralDisk(cd.ContainerDiskFor(cd.ContainerDiskAlpine))
+					vmi_noevict := libvmi.NewAlpine()
```
Comment (xpivarc) [inline tests/migration_test.go:3104]: I think we should make it migratable.

File: tests/migration_test.go [hunk 10]
```diff
@@ -3357,7 +3368,7 @@ var _ = Describe("[rfe_id:393][crit:high][vendor:cnv-qe@redhat.com][level:system
 		var hugepagesVmi *v1.VirtualMachineInstance
 
 		BeforeEach(func() {
-			hugepagesVmi = tests.NewRandomVMIWithEphemeralDisk(cd.ContainerDiskFor(cd.ContainerDiskAlpine))
+			hugepagesVmi = libvmi.NewAlpine()
```
Comment (xpivarc) [inline tests/migration_test.go:3371]: Same as above

File: tests/migration_test.go [hunk 11]
```diff
@@ -3626,12 +3645,15 @@ var _ = Describe("[rfe_id:393][crit:high][vendor:cnv-qe@redhat.com][level:system
...
-			migratableVMI.Spec.Domain.CPU = &v1.CPU{
-				Cores:                 uint32(2),
-				DedicatedCPUPlacement: true,
-			}
-			migratableVMI.Spec.Domain.Resources.Requests[k8sv1.ResourceMemory] = resource.MustParse("512Mi")
+			migratableVMI = libvmi.NewAlpine(
+				libvmi.WithInterface(libvmi.InterfaceDeviceWithMasqueradeBinding()),
+				libvmi.WithNetwork(v1.DefaultPodNetwork()),
+				libvmi.WithCPU(&v1.CPU{
+					Cores:                 uint32(2),
```
Comment (xpivarc) [inline tests/migration_test.go:3652]: No need for `uint32`

**Review by xpivarc — CHANGES_REQUESTED** [review]

File: tests/libvmi/factory.go [hunk 12]
```diff
@@ -60,6 +59,16 @@ func NewCirros(opts ...Option) *kvirtv1.VirtualMachineInstance {
 	return New(RandName(DefaultVmiName), cirrosOpts...)
 }
 
+// NewCirros instantiates a new CirrOS based VMI configuration
+func NewCirros(opts ...Option) *kvirtv1.VirtualMachineInstance {
+	return newCirros(cd.ContainerDiskFor(cd.ContainerDiskCirros), opts...)
+}
+
+// NewCirrosCustomLocation instantiates a new CirrOS custom location based VMI configuration
+func NewCirrosCustomLocation(opts ...Option) *kvirtv1.VirtualMachineInstance {
```
[the remark this answers is not in the document]
Comment (acardace) [reply tests/libvmi/factory.go:62]: Where is thi PR exactly? I cannot find it.

File: tests/migration_test.go [hunk 13]
```diff
@@ -1550,7 +1550,6 @@ var _ = Describe("[rfe_id:393][crit:high][vendor:cnv-qe@redhat.com][level:system
 				vmi.Spec.Domain.Resources.Requests[k8sv1.ResourceMemory] = resource.MustParse(fedoraVMSize)
 				vmi.Spec.Domain.Devices.Rng = &v1.Rng{}
 
-				tests.AddUserData(vmi, "cloud-init", "#!/bin/bash\n echo hello\n")
```
[the remark this answers is not in the document]
Comment (acardace) [reply tests/migration_test.go]: fixed

File: tests/migration_test.go [hunk 14]
```diff
@@ -1639,8 +1639,8 @@ var _ = Describe("[rfe_id:393][crit:high][vendor:cnv-qe@redhat.com][level:system
 				vmi = tests.NewRandomVMIWithPVC(pvName)
 				vmi.Spec.Domain.Resources.Requests[k8sv1.ResourceMemory] = resource.MustParse(fedoraVMSize)
 				vmi.Spec.Domain.Devices.Rng = &v1.Rng{}
-
 				tests.AddUserData(vmi, "cloud-init", "#!/bin/bash\n echo hello\n")
+
```
Comment (EdDev) [inline tests/migration_test.go:1643]: Can you please drop these newline changes? It is confusing.

File: tests/libvmi/factory.go [hunk 15]
```diff
@@ -40,20 +40,29 @@ func NewFedora(opts ...Option) *kvirtv1.VirtualMachineInstance {
 	return New(opts...)
 }
 
-// NewCirros instantiates a new CirrOS based VMI configuration
-func NewCirros(opts ...Option) *kvirtv1.VirtualMachineInstance {
+func newCirros(containerDisk string, opts ...Option) *kvirtv1.VirtualMachineInstance {
```
Comment (EdDev) [inline tests/libvmi/factory.go:43]: This is no longer "Cirros", the name now is not representing what this function creates.
It is Cirros when the result is a VMI that is defined with a Cirros disk. But now the result is any disk.

I suggest you keep the duplication of options, these are two different VMI/s, the fact that they share common options is not enough to enforce a shared function.

File: tests/libvmi/vmi.go [hunk 16]
```diff
@@ -185,6 +185,21 @@ func WithEvictionStrategy(evictionStrategy v1.EvictionStrategy) Option {
 	}
 }
 
+// WithHugePages configures the vmi to use hugepages
+func WithHugePages(hugePagesSize string) Option {
```
Comment (EdDev) [inline tests/libvmi/vmi.go:189]: This seems to leave no room for expanding the `Hugepages` options (without breaking the callers).
(if a new field is added to the struct)

I guess there are two options here:
- Rename to `WithHugePagesSize`.
- Pass in the `Hugepages` struct as input.

File: tests/libvmi/vmi.go [hunk 17]
```diff
@@ -185,6 +185,21 @@ func WithEvictionStrategy(evictionStrategy v1.EvictionStrategy) Option {
...
+	return func(vmi *v1.VirtualMachineInstance) {
+		hugepages := &v1.Hugepages{PageSize: hugePagesSize}
+
+		if vmi.Spec.Domain.Memory != nil {
+			vmi.Spec.Domain.Memory.Hugepages = hugepages
+		} else {
+			vmi.Spec.Domain.Memory = &v1.Memory{
+				Hugepages: &v1.Hugepages{PageSize: hugePagesSize},
+			}
+		}
```
Comment (EdDev) [inline tests/libvmi/vmi.go:199]: What do you think about this option:
```
if vmi.Spec.Domain.Memory == nil {
    vmi.Spec.Domain.Memory = &v1.Memory{}
}
vmi.Spec.Domain.Memory.Hugepages = hugepages
```

File: tests/container_disk_test.go [hunk 18]
```diff
@@ -54,10 +53,8 @@ var _ = Describe("[rfe_id:588][crit:medium][vendor:cnv-qe@redhat.com][level:comp
 		util.PanicOnError(err)
 	})
 
-	verifyContainerDiskVMI := func(vmi *v1.VirtualMachineInstance, obj runtime.Object) {
-		_, ok := obj.(*v1.VirtualMachineInstance)
-		Expect(ok).To(BeTrue(), "Object is not of type *v1.VirtualMachineInstance")
-		tests.WaitForSuccessfulVMIStart(obj)
+	verifyContainerDiskVMI := func(vmi *v1.VirtualMachineInstance) {
```
Comment (EdDev) [inline tests/container_disk_test.go:56]: This change does not seem to be related to the commit.

File: tests/container_disk_test.go [hunk 19]
```diff
@@ -100,19 +97,21 @@ var _ = Describe("[rfe_id:588][crit:medium][vendor:cnv-qe@redhat.com][level:comp
...
 
+					createdVMI := obj.(*v1.VirtualMachineInstance)
 					By("Stopping the VirtualMachineInstance")
-					_, err = virtClient.RestClient().Delete().Resource("virtualmachineinstances").Namespace(vmi.GetObjectMeta().GetNamespace()).Name(vmi.GetObjectMeta().GetName()).Do(context.Background()).Get()
+					_, err = virtClient.RestClient().Delete().Resource("virtualmachineinstances").
+						Namespace(createdVMI.Namespace).Name(createdVMI.Name).Do(context.Background()).Get()
 					Expect(err).To(BeNil())
 					By("Waiting until the VirtualMachineInstance is gone")
-					tests.WaitForVirtualMachineToDisappearWithTimeout(vmi, 120)
+					tests.WaitForVirtualMachineToDisappearWithTimeout(createdVMI, 120)
```
Comment (EdDev) [inline tests/container_disk_test.go:114]: These changes do not seem to be related to the commit.

File: tests/vmi_lifecycle_test.go [hunk 20]
```diff
@@ -1015,7 +1012,10 @@ var _ = Describe("[rfe_id:273][crit:high][arm64][vendor:cnv-qe@redhat.com][level
 				config.CPUModel = defaultCPUModel
 				tests.UpdateKubeVirtConfigValueAndWait(*config)
 
-				newVMI := newCirrosVMI()
+				newVMI := libvmi.NewCirros(
+					libvmi.WithNetwork(v1.DefaultPodNetwork()),
+					libvmi.WithInterface(libvmi.InterfaceDeviceWithMasqueradeBinding()),
```
Comment (EdDev) [inline tests/vmi_lifecycle_test.go:1017]: Is there a reason for needing these network options?

File: tests/vmi_lifecycle_test.go [hunk 21]
```diff
@@ -1033,10 +1033,11 @@ var _ = Describe("[rfe_id:273][crit:high][arm64][vendor:cnv-qe@redhat.com][level
 				config.CPUModel = defaultCPUModel
 				tests.UpdateKubeVirtConfigValueAndWait(*config)
 
-				newVMI := newCirrosVMI()
-				newVMI.Spec.Domain.CPU = &v1.CPU{
-					Model: vmiCPUModel,
-				}
+				newVMI := libvmi.NewCirros(
+					libvmi.WithNetwork(v1.DefaultPodNetwork()),
+					libvmi.WithInterface(libvmi.InterfaceDeviceWithMasqueradeBinding()),
```
Comment (EdDev) [inline tests/vmi_lifecycle_test.go:1038]: Is there a reason for needing these network options?

File: tests/vmi_lifecycle_test.go [hunk 22]
```diff
@@ -979,11 +975,12 @@ var _ = Describe("[rfe_id:273][crit:high][arm64][vendor:cnv-qe@redhat.com][level
...
 
-				vmi := tests.NewRandomVMIWithEphemeralDiskAndUserdata(cd.ContainerDiskFor(cd.ContainerDiskCirros), "#!/bin/bash\necho 'hello'\n")
-				vmi.Spec.Domain.CPU = &v1.CPU{
-					Model: vmiCPUModel,
-				}
-				_, err = virtClient.VirtualMachineInstance(vmi.Namespace).Create(vmi)
+				vmi := libvmi.NewCirros(
+					libvmi.WithCPU(&v1.CPU{
+						Model: vmiCPUModel,
+					}),
```
Comment (EdDev) [inline tests/vmi_lifecycle_test.go:981]: nit: I think this can be a oneliner, like used later on.

File: tests/vmi_lifecycle_test.go [hunk 23]
```diff
@@ -1692,24 +1708,25 @@ var _ = Describe("[rfe_id:273][crit:high][arm64][vendor:cnv-qe@redhat.com][level
...
+					vmi = libvmi.NewCirros(libvmi.WithTerminationGracePeriod(gracePeriod))
 				} else {
 					gracePeriod = v1.DefaultGracePeriodSeconds
+					vmi = libvmi.NewCirros()
 					vmi.Spec.TerminationGracePeriodSeconds = nil
 				}
 
 				By("Creating the VirtualMachineInstance")
-				obj, err := virtClient.VirtualMachineInstance(util.NamespaceTestDefault).Create(vmi)
+				vmi, err := virtClient.VirtualMachineInstance(util.NamespaceTestDefault).Create(vmi)
```
Comment (EdDev) [inline tests/vmi_lifecycle_test.go:1722]: It does not seem related to this commit.

File: tests/libvmi/factory.go [hunk 24]
```diff
@@ -40,20 +40,29 @@ func NewFedora(opts ...Option) *kvirtv1.VirtualMachineInstance {
...
 	return New(cirrosOpts...)
 }
 
+// NewCirros instantiates a new CirrOS based VMI configuration
+func NewCirros(opts ...Option) *kvirtv1.VirtualMachineInstance {
+	return newCirros(cd.ContainerDiskFor(cd.ContainerDiskCirros), opts...)
+}
+
+// NewCirrosCustomLocation instantiates a new CirrOS custom location based VMI configuration
+func NewCirrosCustomLocation(opts ...Option) *kvirtv1.VirtualMachineInstance {
```
Comment (EdDev) [inline tests/libvmi/factory.go:62]: I saw only one single place it is used. I do not think it is worth adding under `libvmi`.
Specialized options that are rarely used and have some logic, should better be placed close to its usage.

File: tests/migration_test.go [hunk 25]
```diff
@@ -2948,25 +2951,25 @@ var _ = Describe("[rfe_id:393][crit:high][vendor:cnv-qe@redhat.com][level:system
 				}
 			}
 
-			getVmisNamespace := func(vmi *v1.VirtualMachineInstance) *k8sv1.Namespace {
```
Comment (EdDev) [inline tests/migration_test.go:2952]: This change does not seem to be related to this commit.

File: tests/migration_test.go [hunk 26]
```diff
@@ -3399,9 +3405,16 @@ var _ = Describe("[rfe_id:393][crit:high][vendor:cnv-qe@redhat.com][level:system
 					tests.StartVirtualMachine(vm_noevict)
 
 					// Get VMIs
-					vmi_evict1, err = virtClient.VirtualMachineInstance(vmi_evict1.Namespace).Get(vmi_evict1.Name, &metav1.GetOptions{})
-					vmi_evict2, err = virtClient.VirtualMachineInstance(vmi_evict1.Namespace).Get(vmi_evict2.Name, &metav1.GetOptions{})
-					vmi_noevict, err = virtClient.VirtualMachineInstance(vmi_evict1.Namespace).Get(vmi_noevict.Name, &metav1.GetOptions{})
+					vmi_evict1, err = virtClient.VirtualMachineInstance(vm_evict1.Namespace).Get(vm_evict1.Name, &metav1.GetOptions{})
+					Expect(err).ToNot(HaveOccurred())
```
Comment (EdDev) [inline tests/migration_test.go:3409]: This change does not seem to be related to this commit.
Same for the other error and NodeName checking below.

File: tests/migration_test.go [hunk 27]
```diff
@@ -3470,8 +3483,8 @@ var _ = Describe("[rfe_id:393][crit:high][vendor:cnv-qe@redhat.com][level:system
 				libnode.AddLabelToNode(sourceNode.Name, cleanup.TestLabelForNamespace(util.NamespaceTestDefault), "target")
 
 				By("starting four VMIs on that node")
-				for _, vmi := range vmis {
-					_, err := virtClient.VirtualMachineInstance(util.NamespaceTestDefault).Create(vmi)
+				for i := range vmis {
+					vmis[i], err = virtClient.VirtualMachineInstance(util.NamespaceTestDefault).Create(vmis[i])
```
Comment (EdDev) [inline tests/migration_test.go:3487]: This change does not seem to be related to this commit.

**Review by EdDev — CHANGES_REQUESTED** [review]

File: tests/libvmi/factory.go (hunk 15 above)
Comment (acardace) [reply tests/libvmi/factory.go:43]: The function is private, and it is just a helper function to remove code duplication.

I can rename it but I think we should not have two functions alike like that.

File: tests/libvmi/vmi.go (hunk 16 above)
Comment (acardace) [reply tests/libvmi/vmi.go]: done, thank you.

File: tests/libvmi/vmi.go (hunk 17 above)
Comment (acardace) [reply tests/libvmi/vmi.go]: done, thank you.

File: tests/container_disk_test.go (hunk 18 above)
Comment (acardace) [reply tests/container_disk_test.go:56]: It's a small fixup that for simplicity I've included in the commit.

File: tests/vmi_lifecycle_test.go (hunk 23 above)
Comment (acardace) [reply tests/vmi_lifecycle_test.go:1722]: It's a small fixup that for simplicity I've included in the commit.

File: tests/migration_test.go (hunk 14 above)
Comment (acardace) [reply tests/migration_test.go:1643]: Why do you think it's confusing? I just divided up the parts where I set up the VMI from the part when I run it, it actually reads better for me.

File: tests/container_disk_test.go (hunk 19 above)
Comment (acardace) [reply tests/container_disk_test.go:114]: They are related, I'm not sure why you're saying they're not, can you elaborate?

File: tests/vmi_lifecycle_test.go (hunk 20 above)
Comment (acardace) [reply tests/vmi_lifecycle_test.go:1017]: nope, good catch, they were there previously but are not needed, thank you.

File: tests/vmi_lifecycle_test.go (hunk 21 above)
Comment (acardace) [reply tests/vmi_lifecycle_test.go:1038]: nope, good catch, they were there previously but are not needed, thank you.

File: tests/vmi_lifecycle_test.go (hunk 22 above)
Comment (acardace) [reply tests/vmi_lifecycle_test.go:981]: changed, thank you.

File: tests/libvmi/factory.go (hunk 15 above)
Comment (acardace) [reply tests/libvmi/factory.go:43]: I removed it since I removed NewCirrosCustomLocation().

File: tests/libvmi/factory.go (hunk 24 above)
Comment (acardace) [reply tests/libvmi/factory.go:62]: Removed it, thank you.

File: tests/migration_test.go (hunk 25 above)
Comment (acardace) [reply tests/migration_test.go:2952]: I had to refactor the test to have it passing, and as a result the function is not needed anymore.

File: tests/migration_test.go (hunk 26 above)
Comment (acardace) [reply tests/migration_test.go:3409]: It's hardening the test, it really does not need another commit I think.

File: tests/migration_test.go (hunk 27 above)
Comment (acardace) [reply tests/migration_test.go:3487]: Yes it's needed to pass the test, otherwise we have no namespace set inside the vmi struct.

File: tests/migration_test.go (hunk 14 above)
Comment (EdDev) [reply tests/migration_test.go:1643]: As a reviewer I am looking for changes related to the intentions described in the commit message. If I see changes, it takes me time to understand that they do nothing. I prefer to see only the things that relate to what is relevant to ease the review. I usually do not mind if it is close to some other code changed, but this one is not.

I also think that it is a more organized to do it in a separate commit, helping future git history readers see only the relevant changes.

File: tests/container_disk_test.go (hunk 18 above)
Comment (EdDev) [reply tests/container_disk_test.go:56]: I think it will be cleaner if you move it to a separate commit.

File: tests/container_disk_test.go [hunk 28]
```diff
@@ -190,7 +186,13 @@ var _ = Describe("[rfe_id:588][crit:medium][vendor:cnv-qe@redhat.com][level:comp
 	Describe("[rfe_id:273][crit:medium][vendor:cnv-qe@redhat.com][level:component]Starting from custom image location", func() {
 		Context("with disk at /custom-disk/downloaded", func() {
 			It("[test_id:1466]should boot normally", func() {
-				vmi := tests.NewRandomVMIWithEphemeralDiskAndUserdata(cd.ContainerDiskFor(cd.ContainerDiskCirrosCustomLocation), "#!/bin/bash\necho 'hello'\n")
+				cirrosOpts := []libvmi.Option{
```
Comment (EdDev) [inline tests/container_disk_test.go:189]: nit: You could inline this.

File: tests/vmi_lifecycle_test.go [hunk 29]
```diff
@@ -1033,10 +1026,7 @@ var _ = Describe("[rfe_id:273][crit:high][arm64][vendor:cnv-qe@redhat.com][level
 				config.CPUModel = defaultCPUModel
 				tests.UpdateKubeVirtConfigValueAndWait(*config)
 
-				newVMI := newCirrosVMI()
-				newVMI.Spec.Domain.CPU = &v1.CPU{
-					Model: vmiCPUModel,
-				}
+				newVMI := libvmi.NewCirros(libvmi.WithCPU(&v1.CPU{Model: vmiCPUModel}))
```
Comment (EdDev) [inline tests/vmi_lifecycle_test.go:1029]: I think this is the right thing to do.
Just note that recently `WithCPUModel` was added which does the same.

File: tests/vmi_lifecycle_test.go (hunk 23 above)
Comment (EdDev) [reply tests/vmi_lifecycle_test.go:1711]: I think that it will help borth the review and the cleanness of the history to place this in a separate commit.

File: tests/migration_test.go (hunk 25 above)
Comment (EdDev) [reply tests/migration_test.go:2952]: ack

File: tests/migration_test.go [hunk 30]
```diff
@@ -2948,25 +2951,25 @@ var _ = Describe("[rfe_id:393][crit:high][vendor:cnv-qe@redhat.com][level:system
...
 
-				vmi := tests.NewRandomVMIWithEphemeralDisk(cd.ContainerDiskFor(cd.ContainerDiskAlpine))
+				vmi := libvmi.NewAlpine(
+					libvmi.WithInterface(libvmi.InterfaceDeviceWithMasqueradeBinding()),
+					libvmi.WithNetwork(v1.DefaultPodNetwork()),
+				)
+
+				namespace, err := virtClient.CoreV1().Namespaces().Get(context.Background(), util.NamespaceTestDefault, metav1.GetOptions{})
+				Expect(err).ShouldNot(HaveOccurred())
+				Expect(namespace).ShouldNot(BeNil())
```
Comment (EdDev) [inline tests/migration_test.go:2967]: nit: Usually we use `ToNot` or `NotTo`.

File: tests/migration_test.go (hunk 26 above)
Comment (EdDev) [reply tests/migration_test.go:3409]: It is an unexpected change, I prefer you move it to a separate commit and explain it is for hardening/fixing.

File: tests/migration_test.go [hunk 31]
```diff
@@ -3071,7 +3074,7 @@ var _ = Describe("[rfe_id:393][crit:high][vendor:cnv-qe@redhat.com][level:system
 			It("[sig-compute][test_id:3243]should recreate the PDB if VMIs with similar names are recreated", func() {
 				for x := 0; x < 3; x++ {
 					By("creating the VMI")
-					_, err := virtClient.VirtualMachineInstance(util.NamespaceTestDefault).Create(vmi)
+					createdVMI, err := virtClient.VirtualMachineInstance(util.NamespaceTestDefault).Create(vmi)
```
Comment (maiqueb) [inline tests/migration_test.go:3077]: is there a reason why you're avoiding using the `vmi` variable here ? Why are you preserving the original VMI spec ?

File: tests/migration_test.go (hunk 26 above)
Comment (maiqueb) [reply tests/migration_test.go:3409]: I agree - the test looks better, and is a lot more deterministic from the added expectations, but this change is unrelated to "using `libvmi.NewAlpine()`"

**Review by maiqueb — COMMENTED** [review]
I am OK with the code, but don't want to lgtm it while @EdDev 's comments are still open.

I think's he's fundamentally right on the "move to other commit" he made throughout the code.

Other than that, code looks good.
